packages/ceti-tag-data-capture/src/sensors/boardTemperature.c

```diff
@@ -74,14 +77,55 @@ void* boardTemperature_thread(void* paramPtr) {
         // Acquire timing and sensor information as close together as possible.
         global_time_us = get_global_time_us();
         rtc_count = getRtcCount();
-        if(getBoardTemperature(&boardTemperature_c) < 0)
+
+//        if(getBoardTemperature(&boardTemperature_c) < 0)        
+        if(getTemperatures(&boardTemperature_c,&batteryTemperature_c) < 0)
           strcat(boardTemperature_data_file_notes, "ERROR | ");
+
         if(boardTemperature_c < -80) // it seems to return -83 when no sensor is connected
         {
           CETI_LOG("XXX readings are likely invalid");
           strcat(boardTemperature_data_file_notes, "INVALID? | ");
         }
+        
+        // ******************   Battery Temperature Checks *************************
+
+        if( (batteryTemperature_c > MAX_CHARGE_TEMP) ||  (batteryTemperature_c < MIN_CHARGE_TEMP) ) {          
+          if (!charging_disabled){  
+            disableCharging();
+            charging_disabled = 1;
```

Comment:
Thanks Michael.  I thought about this and would also like to get Joseph's input.  The thing is, the two flags charging_disabled and discharging_disabled need to be set based on the temperature reading from the temp sensor IC.  This variable currently has file scope in boardTemperature.c, so if I want to set the flags in battery.c somehow, I would need to make the temperature reading variable a global.  I suppose there's nothing wrong with that in principle, but I try to avoid globals where possible as it tends to lead to some spaghetti.

Maybe this would be best done by moving the whole thing into battery.c and decoupling it entirely from boardTemperature.c    If we do that, I would log the battery temperature in the battery.csv, not boardTemperature.csv

Reply:
Looking at this more closely, I agree these flags belong where you have them.

I thought they were indicating that the BMS is already in our desired state to save on i2c interactions (if this was the intent there was no reason for the checks/states to be handled by `boardTemperature.c`). I now realize they are actually indicating that the temperature is currently outside its acceptable range. Maybe a name that reflects this intent (`invalid_charging_temp`, `charging_temp_error`, etc.) would be good?

Reply:
The main reason I introduced these two flags was to have a  one-shot behavior. That way we don't have a stream of redundant log messages and i2c writes to the BMS. For example, this would be a sequence -
* First temperature sample - temperature goes out of bounds - disable charging with i2c write to BMS chip and log the event
* Second sample - temperature still out of bounds - do nothing, we are already disabled and it was logged. No need to do it again.
* Third sample - temperature back in bounds - do nothing. Condition is latched until operator resets. For charging, this makes sense, something has gone seriously wrong and someone needs to look at it before continuing.  For discharging, it doesn't matter because if the tag is in the water and discharging is disabled, it will turn off abruptly and the program halts.